### PR TITLE
Add support for resource-specific tags

### PR DESCRIPTION
**Pull request by awendt**

### CODE REVIEW

**Review by jtsaito — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -21,7 +21,10 @@ resource "aws_s3_bucket" "athena-workspace" {
     }
   }
 
-  tags = var.tags
+  tags = merge(
+    var.tags,
+    lookup(var.resource_specific_tags, "bucket", {})
```
Comment (fmasuhr) [inline main.tf:26]: does it maybe make sense to always use the resource name like this?
```suggestion
    lookup(var.resource_specific_tags, "s3_bucket", {})
```

Comment (awendt) [reply main.tf:26]: @fmasuhr and also this?

```hcl
lookup(var.resource_specific_tags, "athena_workgroup", {})
```

Comment (fmasuhr) [reply main.tf:26]: Right then we change that one as well :)

**Review by fmasuhr — APPROVED** [review]

File: variables.tf [hunk 2]
```diff
@@ -4,6 +4,14 @@ variable "name" {
   description = "Name used for resources and depending resources."
 }
 
+variable "resource_specific_tags" {
+  description = "Map of tags to assign to specific resources supporting tags. Merged with `tags`."
```
Comment (jansiwy) [inline variables.tf:8]: Let's list the supported object types, or use the new optional object attributes right away (https://www.terraform.io/docs/language/expressions/type-constraints.html#experimental-optional-object-type-attributes)

**Pull request merged**